### DISCUSSION TITLE
Some questions about the Iter

### DISCUSSION THREAD

**Discussion by guuzaa** [moonbitlang/core#971] [Q&A] [answered]
The document says `Iter` is a type alias for `((T) -> IterResult) -> IterResult`. And if we want to use `for _ in ...` on a type, we must define an Iter function for it. However, I found a problem in the Array (see the photo).
![image](https://github.com/user-attachments/assets/1207870a-ccc8-4c64-a750-76e2465a2574)
In the implementation, it uses  `for _ in ...`  on `self`, where it refers to the Array. However, we haven't defined any Iter function for it yet. How does it work? Am I mistaken? 
If you could help me, I would appreciate it.

**Comment by Yoorkin** [comment, marked as answer]
The `for..in.. loop` for arrays will be transformed into an index loop. This is specialized by the compiler. Please do not rely on this behavior, it might be changed.
